apache/yunikorn-core
# [YUNIKORN-2766] Only generate event if all predicates failed

### What is this PR for?
Only send an event during scheduling when *all* predicates failed. Otherwise the pod been has assigned to a node successfully.

### What type of PR is it?

### Todos

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2766

### How should this be tested?
1. Unit tests.
2. Create an unschedulable pod (eg. non-matching node selector) and observe the pod events.

### Screenshots (if appropriate)

### Questions:
